eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/AssignedOnceVariableToBeFinalCheck.java

```diff
@@ -0,0 +1,239 @@
+    private static Set<String> assignmentsFromConstructor;
+    private static Set<String> assignmentsFromMethods;
+    private static DetailAST scopeParent = null;
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.CLASS_DEF,
+            TokenTypes.ASSIGN,
+            TokenTypes.PLUS_ASSIGN,
+            TokenTypes.BAND_ASSIGN,
+            TokenTypes.BOR_ASSIGN,
+            TokenTypes.BSR_ASSIGN,
+            TokenTypes.BXOR_ASSIGN,
+            TokenTypes.DIV_ASSIGN,
+            TokenTypes.MINUS_ASSIGN,
+            TokenTypes.MOD_ASSIGN,
+            TokenTypes.SL_ASSIGN,
+            TokenTypes.SR_ASSIGN,
+            TokenTypes.STAR_ASSIGN,
+            TokenTypes.INC,
+            TokenTypes.POST_INC,
+            TokenTypes.DEC,
+            TokenTypes.POST_DEC,
+            TokenTypes.METHOD_DEF,
+            TokenTypes.CTOR_DEF,
+        };
+    }
+
+    @Override
+    public void beginTree(DetailAST root) {
+        nonFinalFields = new ArrayList<>();
+        assignmentsFromConstructor = new HashSet<>();
+        assignmentsFromMethods = new HashSet<>();
+    }
+
+    @Override
+    public void visitToken(DetailAST token) {
+        switch (token.getType()) {
+            case TokenTypes.CLASS_DEF:
+                fillClassFieldDefinitions(token);
+                break;
+            case TokenTypes.ASSIGN:
+            case TokenTypes.PLUS_ASSIGN:
+            case TokenTypes.BAND_ASSIGN:
+            case TokenTypes.BOR_ASSIGN:
+            case TokenTypes.BSR_ASSIGN:
+            case TokenTypes.BXOR_ASSIGN:
+            case TokenTypes.DIV_ASSIGN:
+            case TokenTypes.MINUS_ASSIGN:
+            case TokenTypes.MOD_ASSIGN:
+            case TokenTypes.SL_ASSIGN:
+            case TokenTypes.SR_ASSIGN:
+            case TokenTypes.STAR_ASSIGN:
+            case TokenTypes.INC:
+            case TokenTypes.POST_INC:
+            case TokenTypes.DEC:
+            case TokenTypes.POST_DEC:
+                checkAssignation(token);
+                break;
+            case TokenTypes.METHOD_DEF:
+                scopeParent = token;
+            case TokenTypes.CTOR_DEF:
+                scopeParent = token;
+            default:
+                // Checkstyle complains if there's no default block in switch
+                break;
+        }
+    }
+
+    @Override
+    public void leaveToken(DetailAST token) {
+        switch (token.getType()) {
+            case TokenTypes.METHOD_DEF:
+            case TokenTypes.CTOR_DEF:
+                scopeParent = null;
+            default:
+                break;
+        }
+    }
+
+    @Override
+    public void finishTree(DetailAST token) {
+        for (DetailAST field: nonFinalFields) {
```

Comment:
nit: missing space before `:`

Reply:
tks. done

Reply:
I'm surprised.. I thought checkstyles was run on this. maybe not.

Reply:
@conniey , I think in the past checkstyle project was child project of pom-client, and in this case the custom implementations for checkstyle rule were checked.
But now it is a plug-in dependency, so it is only compiled as is and use it to check pom-client projects